src/test/java/com/box/sdk/BoxFileTest.java

```diff
@@ -1394,6 +1394,52 @@ public void testUploadNewVersionReturnsCorrectInfo() throws IOException {
         Assert.assertEquals(fileName, info.getName());
     }
 
+    @Test
+    @Category(UnitTest.class)
+    public void createSharedLinkWithPasswordSucceeds() throws IOException {
+        final String fileID = "1111";
+        final String password = "test1";
+        String result = "";
+
+        JsonObject innerObject = new JsonObject()
+                .add("password", password);
+
+        JsonObject sharedLinkObject = new JsonObject()
+                .add("shared_link", innerObject);
+
+        result = TestConfig.getFixture("BoxFile/CreateSharedLinkWithPassword201");
+
+        WIRE_MOCK_CLASS_RULE.stubFor(WireMock.put(WireMock.urlPathEqualTo("/files/" + fileID))
+                .willReturn(WireMock.aResponse()
+                        .withHeader("Content-Type", "application/json")
+                        .withStatus(200)));
+
+        WIRE_MOCK_CLASS_RULE.stubFor(WireMock.put(WireMock.urlPathEqualTo("/files/" + fileID))
+                .willReturn(WireMock.aResponse()
+                        .withHeader("Content-Type", "application/json")
+                        .withBody(result)));
+
+        WIRE_MOCK_CLASS_RULE.stubFor(WireMock.put(WireMock.urlPathEqualTo("/files/" + fileID))
+                .withRequestBody(WireMock.equalToJson(sharedLinkObject.toString()))
+                .willReturn(WireMock.aResponse()
+                        .withHeader("Content-Type", "application/json")
+                        .withBody(result)));
+
+        BoxFile file = new BoxFile(this.api, fileID);
+        BoxSharedLink.Permissions permissions = new BoxSharedLink.Permissions();
+
+        permissions.setCanDownload(true);
+        permissions.setCanPreview(true);
+        BoxSharedLink sharedLink = file.createSharedLink(BoxSharedLink.Access.OPEN, null, permissions);
```

Comment:
Is it possible to allow setting the password in this call instead?  It would be much nicer for users to not have to make two API calls.